src/mbgl/text/cross_tile_symbol_index.cpp

```diff
@@ -99,11 +111,20 @@ bool CrossTileSymbolLayerIndex::addBucket(const OverscaledTileID& tileID, Symbol
     return true;
```

Comment:
would it might make sense to move this `indexes[tileID.overscaledZ].erase(tileID);` up next to where `removeBucketCrossTileIDs(tileID.overscaledZ, previousIndex->second);` is called?

Reply:
Don't we have to leave the tile in the indexes until after we've called `findMatches` in order to copy the values over? The comment "The old index entries themselves stick around until 'removeStaleBuckets' is called." is misleading though, because in this case they only stick around until the end of `addBucket`.

Reply:
Yep, you're right